python/ray/tune/execution/ray_trial_executor.py

```diff
@@ -310,43 +316,79 @@ def _stage_and_update_status(self, trials: Iterable[Trial]):
 
         Stages placement groups of all trials.
         """
-        if not self._has_cleaned_up_pgs:
-            # Clean up existing placement groups after trigger the tuning
-            # run step() method for the first time
-            self._pg_manager.cleanup_existing_pg()
-            self._has_cleaned_up_pgs = True
-
         for trial in trials:
+            if len(self._staged_trials) >= self._max_staged_actors:
+                break
+
             if trial.status not in (Trial.PENDING, Trial.PAUSED):
                 continue
+
             if trial in self._staged_trials:
```

Comment:
seems like what is needed is just to keep a number of staged trials for each resource type?

Reply:
Hm yeah that should work. I can give it a try. Per default I think it's ok to keep it as is as it's existing logic

Reply:
See above, we use staged_trials in other locations, and this way we avoid linearly scanning through all resource requirements on every check. This will likely be refactored soon so let's keep it for now